haystack/document_stores/pinecone.py

```diff
@@ -188,17 +198,8 @@ def __init__(
 
         super().__init__()
 
-    def _add_local_ids(self, index: str, ids: list):
-        """
-        Add all document IDs to the set of all IDs.
-        """
-        if index not in self.all_ids:
-            self.all_ids[index] = set()
-        self.all_ids[index] = self.all_ids[index].union(set(ids))
-
-    def _index_name(self, index) -> str:
-        index = _sanitize_index_name(index) or self.index
-        # self.index = index  # TODO maybe not needed
+    def _index(self, index) -> str:
```

Comment:
why the change from `_index_name` to `_index`? I assumed people when reading may read `index` as being the actual `Index` object rather than the index name string? Minor question, shouldn't slow down merging

Reply:
I agree with you that make more sense to be index_name instead of index, since it is not reference to Index object. 

But, this change is made just in order to be consistent in the naming with the attribute on class level, since in the constructor of `PineconeDocumentStore` class (as well as base class), `index` parameter indicates the name of the index, not `Index` reference.